### ISSUE TITLE
Numerical Errors in segmentation of midi files

### ISSUE THREAD

**Issue by cedricviaccoz** [DCMLab/wavescapes#2]
Getting the PCV containing the pitch content of the first 4 measures from Bach's Prelude in C Major through the function "produce_pitch_class_matrix_from_filename" returns a vectors whose value seem to be inconsistent with the Prelude's Score:

```python
>>> wavescape.produce_pitch_class_matrix_from_filename('BachPrelude.mid', aw_size=16.)[0]
array([14.        ,  0.        ,  8.66666667,  0.        ,  8.66666667,
        2.        ,  0.        ,  3.        ,  0.        ,  1.        ,
        0.        ,  4.        ])
```

Counts for pitch-classes D and E (8.6666667) are obviously false for some reason. Considering they are the pitch class of dotted-eight notes, the issue might be coming from the way the program counts dotted notes.

**Comment by cedricviaccoz** [comment]
UPDATE: on the full Prelude, getting the PCV from the first 4 bars returns the right result (`array([14.,  0.,  9.,  0.,  9.,  2.,  0.,  3.,  0.,  1.,  0.,  4.])`), the above bogus behavior arises when the file given is a midi file holding only those first 4 measures

**Comment by fabianmoss** [comment]
Can it have to do with the window size or the resolution? The overall relations seem to make sense (mostly C, E, and G)

**Comment by cedricviaccoz** [comment]
Considering I was able to get the right amount of counts with the original file with the same exact parameters I don't think so. My guess would be that it's rather an issue that comes from the way Musescore export MIDI file. 

Interestingly, multiplying the value `8.6666667` by 3 gives a value that is really close to 26, and the expected value for the count of C and D is 27 divided by 3.

**Comment by fabianmoss** [comment]
Precision is a bitch

**Comment by robert-lieck** [comment]
Just came across this... If you export piano scores from musescore to midi, the notes will be shorter than their symbolic duration (for some other instruments this is different). That probably is to make it sound better. Took me half a day to find out because I thought it's a bug in music21 or my code. But this might be what you're observing here, too.

**Comment by cedricviaccoz** [comment]
Thanks for chiming in Robert, considering how I created the four bars version of the Prelude, this is likely what happpened to me as well. Do you know if there is a setting or a way to prevent MuseScore from truncating symbolic durations?

**Comment by robert-lieck** [comment]
Nope, there might be but I don't know. So far I decided to not care about these deviations because they are relatively small. But otherwise I would use music XML instead of MIDI, which would only require using a different reader (integrating a music XML reader into the corpus interface is on the ToDo list for some time now).
